### ISSUE TITLE
(Linux) Keyboard downListener isn't called for number keys when Shift is held

### ISSUE THREAD

**Issue by AlexRamallo** [Kode/Kore#480]
I tested this on Windows with Krom and Linux with Krom and native, and the issue only appears on my Linux machine (running Ubuntu/KDE on X11).

**The Issue:**
If you attempt to input the character '@' on a standard US keyboard, you need to hold the Shift key and press the number 2.

 * On Windows, this results in Krom calling the `onKeyDown` listener with key code `50`, and `onKeyPress` with the character '@'

 * On Linux, this results in Krom only calling `onKeyPress` with the character '@', but not `onKeyDown`

**Sample code:**
```
import kha.*;
import kha.input.*;

class Main {

	public static function main() {
		var opt = new kha.System.SystemOptions("Test", 800, 600);
		System.start(opt, onStart);
	}

	public static function onStart(win:Window){
		Assets.loadEverything(function(){
			Keyboard.get().notify(onKeyDown, onKeyUp, onKeyPress);
		});
	}

	public static function onKeyDown(code:KeyCode){
		trace('onKeyDown: $code');
	}

	public static function onKeyUp(code:KeyCode){
		trace('onKeyUp: $code');
	}

	public static function onKeyPress(char:String){
		trace('onKeyPress: $char');
	}
}
```

**Comment by AlexRamallo** [comment]
Looks like the problem is because the X event handler is switching on the `keysym` obtained from `XLookupString`. ([Line 513 here](https://github.com/Kode/Kinc/blob/master/Backends/System/Linux/Sources/Kore/System.cpp#L513)). Since `XLookupString` returns the character value rather than the key code, holding the Shift key will modify the returned value and it won't match with the `XK_2` case (or others).

It looks like whoever wrote that part initially also tried to account for this by adding duplicate cases for letter keys, like:

```
KEY(XK_a, KINC_KEY_A)
KEY(XK_A, KINC_KEY_A)
```

But that solution would totally fall apart for non-letter keys when you add different keyboard encodings.

I think an appropriate solution would be to obtain the `KeySym` for the event twice: once for the `key_press` event and again for the `key_down` event, but ignoring the shift modifier.

For example, by replacing line 513 (linked above) with:
```
KeySym ksKey = XkbKeycodeToKeysym(Kore::Linux::display, event.xkey.keycode, 0, 0);
switch (ksKey) {
```

That fixes the problem, and also enables the removal of those duplicate cases.


Also it might be helpful to know that the reason I discovered this problem was because I was using Zui (https://github.com/armory3d/zui) in a project, and I noticed that on Linux I was not able to enter symbols like '@' into text fields, but I could on Windows. It's possible to work around that with a different implementation of input handling from Zui, but this fixes the underlying issue
